Nasal/light-manager.nas

```diff
@@ -0,0 +1,303 @@
+# provides relative vectors from eye-point to aircraft lights
+# in east/north/up coordinates the renderer uses
+
+var light_manager = {
+    run: 0,
+
+    lat_to_m: 110952.0,
+    lon_to_m: 0.0,
+
+    light1_xpos: 0.0,
```

Comment:
Not necessarily, the layout is a standard c++ class layout with all the member variables and functions defined. When you init() you are basically making a "new" class x with all the vars and functions set and ready to go. Then you change any vars that need a different starting value than the default and off you go.
In c++ you have to do it like this or you will get an error. In Lua you can make the member var for the first time in a member function (init).
In this case if we don't create all the member var's where they are being initiated, we have to make sure any that are used are initialized in the init section, even those that are set to 0.
I went through and set it like that and there were a few that are not changed in the init section that are required and would have to be added to the init section and set to 0. The way it is set up now is the correct way to do it. But you are amplifying one of the reasons some experienced programmers don't care for c++ class structure. I have evolved over time to not like programming with classes. It carries more overhead than is ever needed.